### PR TITLE
Don't allow users to submit Android webcompat reports from browser error pages

### PR DESCRIPTION
**Pull request by vadimstruts**
Resolves https://github.com/brave/brave-browser/issues/36484

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
1. Go to https://badssl.com/ and go to any of the SSL error pages like `expired`
2. Open Brave Shields pop up dialog.
3. Switch off Brave Shields if It is ON.
4. Click "Report broken site".
5. Make sure that you can;t send report for error page

### CODE REVIEW

File: browser/ui/android/strings/android_brave_strings.grd [hunk 1]
```diff
@@ -1621,6 +1621,9 @@ Are you sure you want to do this?
       <message name="IDS_REPORT_BROKEN_SITE_TEXT_2" desc="Shields text for broken text warning message.">
         Note: This site address will be submitted with your Brave version number and your IP address(which will not be stored).
       </message>
+      <message name="IDS_REPORT_ERROR_PAGE_EXPLANATION" desc="Notifies the user that the error page cannot be reported">
+        This page is an error page and is not eligible for reporting.
```
Comment (mkarolin) [inline browser/ui/android/strings/android_brave_strings.grd:1625]: Maybe simplify a bit: "This is an error page. It is not eligible for reporting." cc: @rmcfadden3

Comment (rmcfadden3) [reply browser/ui/android/strings/android_brave_strings.grd:1625]: Maybe like `This is an error page. It cannot be reported as a broken site.`

File: android/java/res/layout/report_error_page_layout.xml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        android:layout_marginBottom="14dp"
+        android:text="@string/report_error_page_explanation"/>
+
+    <android.widget.Button
+        android:id="@+id/btn_close"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_gravity="center"
+        android:background="@drawable/transparent_bg_bordered"
+        android:paddingTop="8dp"
```
Comment (deeppandya) [inline android/java/res/layout/report_error_page_layout.xml:39]: nit : we can use android:paddingHorizontal=8dp" and android:paddingVertical="16dp"

**Review by deeppandya — APPROVED** [review]
LGTM

**Review by mkarolin — APPROVED** [review]
`strings`++

File: android/java/res/layout/report_error_page_layout.xml (hunk 2 above)
Comment (vadimstruts) [reply android/java/res/layout/report_error_page_layout.xml]: done

**Pull request merged**